src/staged_policy.jl

```diff
@@ -0,0 +1,47 @@
+
+"""
+StagedPolicy
+
+Policy for finite horizon POMDP solvers.
+Wraps multiple Policies, one for each stage.
+"""
+struct StagedPolicy{M<:Union{POMDP,MDP}, P<:Policy}<:Policy
+    pomdp::M
+    staged_policies::Array{P, 1}
+
+    function StagedPolicy(m::M, p::Vector{P}) where {M<:Union{POMDP,MDP}, P<:Policy}
+        typeof(HorizonLength(m)) == InfiniteHorizon ? throw(ArgumentError("Argument pomdp should be
+                             valid Finite Horizon POMDP with methods from FiniteHorizonPOMDPs.jl
+                             interface implemented. If you are completely sure that you implemented
+                             all of them, you should also check if you have defined
+                             HorizonLength(::Type{<:MyFHMDP})")) : new{M, P}(m, p)
+     end
+end
+
+### functional methods ###
+value(p::StagedPolicy, b, t) = t <= horizon(p.pomdp) ? value(p.staged_policies[t], b) : 0.
+
+action(p::StagedPolicy, b, t) = t <= horizon(p.pomdp) ? action(p.staged_policies[t], b) : ordered_actions(p.pomdp)[1]
+
+# Given that, for example, POMDPSimulators.jl does not work with action(p, b, t),
+# we need to somehow introduce the stage variable
+action(p::StagedPolicy, b::DiscreteBelief) = action(p, b, only(unique(map(stage, p.pomdp, b.state_list))))
```

Comment:
Actually, now that I think about this a little more, it won't work correctly since `b.state_list` is all of the states in the POMDP. 

It should be possible to check through all of the nonzero probaility states without allocating with something like this:
```suggestion
function action(p::StagedPolicy, b::DiscreteBelief)
    t = -1
    for i in 1:length(b.b)
        if b.b[i] > 0.0
            if t < 0
                t = stage(b.state_list[i])
            elseif t != stage(b.state_list[i])
                error("Not all states in the belief had the same stage")
            end
        end
    end
    return action(p, b, t)
end

```
 
Also, this was not tested - see the test below.